### ISSUE TITLE
COCO Dataset training for some classes only

### ISSUE THREAD

**Issue by fazer125** [ultralytics/yolov5#4454]
## ❔Question
Hello!
I have a question about retraining in COCO Dataset.

For example, you want to detect only PERSON, CAT, and DOG.
You can ignore the other classes, but in general, the fewer detection classes you have, the higher the accuracy.

So, is there any way to ignore the labels other than PERSON, CAT and DOG in COCO Dataset and re-train it?

English isn't my first language, so sorry if it's hard to understand!
Thanks!

**Comment by glenn-jocher** [comment]
@fazer125 you can output a subset of classes quite simply, no retraining required. See PyTorch Hub tutorial:

### Inference Settings
Inference settings such as **confidence threshold**, NMS **IoU threshold**, and **classes** filter are model attributes, and can be modified by:
```python
model.conf = 0.25  # confidence threshold (0-1)
model.iou = 0.45  # NMS IoU threshold (0-1)
model.classes = None  # (optional list) filter by class, i.e. = [0, 15, 16] for persons, cats and dogs

results = model(imgs, size=320)  # custom inference size
```

## YOLOv5 Tutorials

* [Train Custom Data](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data)&nbsp; 🚀 RECOMMENDED
* [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/)&nbsp; ☘️ RECOMMENDED
* [Weights & Biases Logging](https://github.com/ultralytics/yolov5/issues/1289)&nbsp; 🌟 NEW
* [Supervisely Ecosystem](https://docs.ultralytics.com/yolov5/tutorials/model_export8)&nbsp; 🌟 NEW
* [Multi-GPU Training](https://docs.ultralytics.com/yolov5/tutorials/multi_gpu_training)
* [PyTorch Hub](https://docs.ultralytics.com/yolov5/tutorials/pytorch_hub_model_loading)&nbsp; ⭐ NEW
* [TorchScript, ONNX, CoreML Export](https://docs.ultralytics.com/yolov5/tutorials/model_export) 🚀
* [Test-Time Augmentation (TTA)](https://docs.ultralytics.com/yolov5/tutorials/test_time_augmentation)
* [Model Ensembling](https://docs.ultralytics.com/yolov5/tutorials/model_ensembling)
* [Model Pruning/Sparsity](https://docs.ultralytics.com/yolov5/tutorials/model_pruning_and_sparsity)
* [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution)
* [Transfer Learning with Frozen Layers](https://docs.ultralytics.com/yolov5/tutorials/transfer_learning_with_frozen_layers)&nbsp; ⭐ NEW
* [TensorRT Deployment](https://github.com/wang-xinyu/tensorrtx)

**Comment by fazer125** [comment]
@glenn-jocher  Okay.
It's not exactly the answer I was looking for, but I got it.
Thanks!

**Comment by ghost** [comment]
@fazer125 could you please tell how you have done it ? 
Thankyou

**Comment by fazer125** [comment]
> @fazer125 could you please tell how you have done it ?
> Thankyou

Using the COCO API, you can get only the classes you want.

**Comment by fazer125** [comment]
@richard2611 
Using the COCO API, you can get only the classes you want.
